src/tests/Interop/common/ComHelpers.h

```diff
@@ -50,20 +52,25 @@ namespace Internal
         return __QueryInterfaceImpl(riid, ppvObject, remain...);
     }
 }
+ #endif  
 
 // Implementation of IUnknown operations
 class UnknownImpl
 {
-public:
-    UnknownImpl() = default;
+public:  
+    UnknownImpl() : _refCount{ 1 } {};
     virtual ~UnknownImpl() = default;
 
     UnknownImpl(const UnknownImpl&) = delete;
     UnknownImpl& operator=(const UnknownImpl&) = delete;
 
-    UnknownImpl(UnknownImpl&&) = default;
+    UnknownImpl(UnknownImpl&&)  : _refCount{ 1 } {};
+ #ifdef _WIN32   
```

Comment:
Can we delete this for all platforms? If not, can we make it `default` for all platforms?

Reply:
I think we have to delete both:
`    UnknownImpl(UnknownImpl&&) = delete;
    UnknownImpl& operator=(UnknownImpl&&) = delete;`

Otherwise, setting default on both I get these errors:

/runtime/src/tests/Interop/common/ComHelpers.h/runtime/src/tests/Interop/common/ComHelpers.h::6767::55: :error:  explicitly defaulted move constructor is implicitly deleted [-Werror,-Wdefaulted-function-deleted]error
: explicitly defaulted move constructor is implicitly deleted [-Werror,-Wdefaulted-function-deleted]
    UnknownImpl(UnknownImpl&&) = default;
    ^
    UnknownImpl(UnknownImpl&&) = default;
    ^
/runtime/src/tests/Interop/common/ComHelpers.h:123:24: note: move constructor of 'UnknownImpl' is implicitly deleted because field '_refCount' has a deleted move constructor
/runtime/src/tests/Interop/common/ComHelpers.h:123:24:     std::atomic<ULONG> _refCount;  note
:                        ^move constructor of 'UnknownImpl' is implicitly deleted because field '_refCount' has a deleted move constructor

/opt/rh/devtoolset-7/root/usr/lib/gcc/x86_64-redhat-linux/7/../../../../include/c++/7/atomic:691:7: note: 'atomic' has been explicitly marked deleted here
    std::atomic<ULONG> _refCount;
                       ^
      atomic(const atomic&) = delete;
/opt/rh/devtoolset-7/root/usr/lib/gcc/x86_64-redhat-linux/7/../../../../include/c++/7/atomic      ^
:691:7: note: 'atomic' has been explicitly marked deleted here
In file included from /runtime/src/tests/Interop/COM/ComWrappers/WeakReference/WeakReferenceNative.cpp:5:
/runtime/src/tests/Interop/common/ComHelpers.h:68:18      atomic(const atomic&) = delete;:
       ^error:
explicitly defaulted move assignment operator is implicitly deleted [-Werror,-Wdefaulted-function-deleted]
    UnknownImpl& operator=(UnknownImpl&&) = default;In file included from
/runtime/src/tests/Interop/COM/ComWrappers/MockReferenceTrackerRuntime/ReferenceTrackerRuntime.cpp                 ^:
5:
/runtime/src/tests/Interop/common/ComHelpers.h:68:18: /runtime/src/tests/Interop/common/ComHelpers.herror:: 123explicitly defaulted move assignment operator is implicitly deleted [-Werror,-Wdefaulted-function-deleted]:
24: note: move assignment operator of 'UnknownImpl' is implicitly deleted because field '_refCount' has a deleted move assignment operator
    UnknownImpl& operator=(UnknownImpl&&) = default;
                 ^    std::atomic<ULONG> _refCount;

                       ^
/opt/rh/devtoolset-7/root/usr/lib/gcc/x86_64-redhat-linux/7/../../../../include/c++/7/atomic:692/runtime/src/tests/Interop/common/ComHelpers.h::15123:: 24note::  'operator=' has been explicitly marked deleted herenote
: move assignment operator of 'UnknownImpl' is implicitly deleted because field '_refCount' has a deleted move assignment operator
    std::atomic<ULONG> _refCount;        atomic& operator=(const atomic&) = delete;

                       ^
              ^
/opt/rh/devtoolset-7/root/usr/lib/gcc/x86_64-redhat-linux/7/../../../../include/c++/7/atomic:692:15: note: 'operator=' has been explicitly marked deleted here
      atomic& operator=(const atomic&) = delete;